### ISSUE TITLE
Tensor Disappear.

### ISSUE THREAD

**Issue by zhyzhyzhy123** [Lightning-AI/pytorch-lightning#19712]
### Bug description

I met a very strange bug, the things is when I train a model, every thing is ok. But when I evaluate the model, where I apply the `trainer.test(model, test_dataloaders=val0_dataloader)`, strange things happened. After the tensor go througt the specefic module, which is an `nn.Conv3d`, the tensor become strange. before tensor get into the module, when print it in vscode debug mode, it appear like this:
![1d62e419620db2bafcc1603e2162b42](https://github.com/Lightning-AI/pytorch-lightning/assets/31156721/bfaffd85-4272-4346-afba-cac3d7e98268)
but after the code:
`        for i in range(1, len(self.conv_list)):
            y += self.bn2[i](self.conv2[i](self.relu(self.bn1[i](self.conv1[i](x_in)))))`
the tensor become like:
![3697c878fccd7303ca93c45c3372c40](https://github.com/Lightning-AI/pytorch-lightning/assets/31156721/92d473e5-39b1-4d8a-bf20-2576800fa09e)
they are the same tensor, but x_in becoming unaccessable.

the version I used is 1.4.9

### What version are you seeing the problem on?

v1.8

### How to reproduce the bug

```python
from monoscene.data.Indoors_v2.indoorsv2_dm import IndoorV2DataModule

from torch.utils.data.dataloader import DataLoader
from monoscene.models.monoscene import MonoScene
from pytorch_lightning import Trainer
from pytorch_lightning.loggers import TensorBoardLogger
from pytorch_lightning.callbacks import ModelCheckpoint, LearningRateMonitor
from pytorch_lightning import seed_everything
import os
#os.environ['CUDA_LAUNCH_BLOCKING'] = "1"
import hydra
from omegaconf import DictConfig
import numpy as np
import torch

@hydra.main(config_name="../../config/indoorv2.yaml")
def main(config: DictConfig):
    torch.set_grad_enabled(False)
    
    full_scene_size = (120,120,72)
    project_scale = 2
    feature = 200
    n_classes = 2
    voxel_size = 0.04
    #class_weights = NYU_class_weights
    data_module = IndoorV2DataModule(
        frustum_size=config.frustum_size,
        batch_size=1,
        num_workers=3,
    )   
    trainer = Trainer(
        sync_batchnorm=True, deterministic=True, gpus=1, accelerator="ddp"
    )

    model_path = os.path.expanduser("~/projects/MonoScene/training_log/indoorsv2/exp_indoorv2_KITTIFLOSP_KITTIDECODER_3_FrusSize_8_nRelations2_WD0.0001_lr0.0001_CEssc_geoScalLoss_fpLoss_CERel_3DCRP_Proj_2_4_8/checkpoints/epoch=000-val/mIoU=0.03209.ckpt")

    model = MonoScene.load_from_checkpoint(
        model_path,
        feature=feature,
        project_scale=project_scale,
        fp_loss=config.fp_loss,
        full_scene_size=full_scene_size,
    )
    model.eval()
    data_module.setup()
    
    print("================Test For Train set================")
    val0_dataloader = data_module.train_dataloader()
    trainer.test(model, test_dataloaders=val0_dataloader)
    #return
    print("================Test For val set================")
    val0_dataloader = data_module.val_dataloader()
    trainer.test(model, test_dataloaders=val0_dataloader)
       
    print("================Test For test set================")
    val0_dataloader = data_module.test_dataloader()
    trainer.test(model, test_dataloaders=val0_dataloader)


if __name__ == "__main__":
    main()


```python
class SegmentationHead(nn.Module):
    """
    3D Segmentation heads to retrieve semantic segmentation at each scale.
    Formed by Dim expansion, Conv3D, ASPP block, Conv3D.
    Taken from https://github.com/cv-rits/LMSCNet/blob/main/LMSCNet/models/LMSCNet.py#L7
    """

    def __init__(self, inplanes, planes, nbr_classes, dilations_conv_list):
        super().__init__()

        # First convolution
        self.conv0 = nn.Conv3d(inplanes, planes, kernel_size=3, padding=1, stride=1)

        # ASPP Block
        self.conv_list = dilations_conv_list
        self.conv1 = nn.ModuleList(
            [
                nn.Conv3d(
                    planes, planes, kernel_size=3, padding=dil, dilation=dil, bias=False
                )
                for dil in dilations_conv_list
            ]
        )
        self.bn1 = nn.ModuleList(
            [nn.BatchNorm3d(planes) for dil in dilations_conv_list]
        )
        self.conv2 = nn.ModuleList(
            [
                nn.Conv3d(
                    planes, planes, kernel_size=3, padding=dil, dilation=dil, bias=False
                )
                for dil in dilations_conv_list
            ]
        )
        self.bn2 = nn.ModuleList(
            [nn.BatchNorm3d(planes) for dil in dilations_conv_list]
        )
        self.relu = nn.ReLU()

        self.conv_classes = nn.Conv3d(
            planes, nbr_classes, kernel_size=3, padding=1, stride=1
        )

    def forward(self, x_in):

        # Convolution to go from inplanes to planes features...
        x_in = self.relu(self.conv0(x_in))

        y = self.bn2[0](self.conv2[0](self.relu(self.bn1[0](self.conv1[0](x_in)))))
        for i in range(1, len(self.conv_list)):
            y += self.bn2[i](self.conv2[i](self.relu(self.bn1[i](self.conv1[i](x_in)))))
        x_in = self.relu(y + x_in)  # modified

        x_in = self.conv_classes(x_in)

        return x_in
```
the question is appear in `self.conv1[i](x_in)`
```


### Error messages and logs

```
# Error messages and logs here please
```
================Test For Train set================
/home/zhuhongyi/anaconda3/envs/monoscene/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py:679: LightningDeprecationWarning: `trainer.test(test_dataloaders)` is deprecated in v1.4 and will be removed in v1.6. Use `trainer.test(dataloaders)` instead.
  "`trainer.test(test_dataloaders)` is deprecated in v1.4 and will be removed in v1.6."
initializing ddp: GLOBAL_RANK: 0, MEMBER: 1/1
[2024-03-29 16:45:04,421][torch.distributed.distributed_c10d][INFO] - Added key: store_based_barrier_key:1 to store for rank: 0
[2024-03-29 16:45:04,422][torch.distributed.distributed_c10d][INFO] - Rank 0: Completed store-based barrier for key:store_based_barrier_key:1 with 1 nodes.
----------------------------------------------------------------------------------------------------
distributed_backend=nccl
All DDP processes registered. Starting ddp with 1 processes
----------------------------------------------------------------------------------------------------

LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
/home/zhuhongyi/anaconda3/envs/monoscene/lib/python3.7/site-packages/pytorch_lightning/trainer/data_loading.py:377: UserWarning: Your test_dataloader has `shuffle=True`, it is best practice to turn this off for val/test/predict dataloaders.
  f"Your {mode}_dataloader has `shuffle=True`, it is best practice to turn"
Testing: 0it [00:00, ?it/s]Floating point exception (core dumped)

### Environment

<details>
  <summary>Current environment</summary>

* CUDA:
        - GPU:
                - NVIDIA GeForce RTX 3090
                - NVIDIA GeForce RTX 3090
                - NVIDIA GeForce RTX 3090
                - NVIDIA GeForce RTX 3090
                - NVIDIA GeForce RTX 3090
                - NVIDIA GeForce RTX 3090
                - NVIDIA GeForce RTX 3090
                - NVIDIA GeForce RTX 3090
        - available:         True
        - version:           11.7
* Lightning:
        - pytorch-lightning: 1.4.9
        - torch:             1.13.1
        - torchmetrics:      0.6.0
        - torchvision:       0.14.1
* Packages:
        - absl-py:           2.0.0
        - aiohttp:           3.8.5
        - aiosignal:         1.3.1
        - antlr4-python3-runtime: 4.8
        - anyio:             3.7.1
        - apptools:          5.2.1
        - argon2-cffi:       23.1.0
        - argon2-cffi-bindings: 21.2.0
        - async-timeout:     4.0.3
        - asynctest:         0.13.0
        - attrs:             23.1.0
        - backcall:          0.2.0
        - beautifulsoup4:    4.12.2
        - bleach:            6.0.0
        - cachetools:        5.3.1
        - certifi:           2022.12.7
        - cffi:              1.15.1
        - charset-normalizer: 3.2.0
        - comm:              0.1.4
        - configobj:         5.0.8
        - cycler:            0.11.0
        - debugpy:           1.7.0
        - decorator:         5.1.1
        - defusedxml:        0.7.1
        - entrypoints:       0.4
        - envisage:          7.0.3
        - exceptiongroup:    1.1.3
        - fastjsonschema:    2.18.0
        - fonttools:         4.38.0
        - frozenlist:        1.3.3
        - fsspec:            2023.1.0
        - future:            0.18.3
        - gdclmscnet:        0.0.0
        - google-auth:       2.23.0
        - google-auth-oauthlib: 0.4.6
        - grpcio:            1.58.0
        - hydra-core:        1.0.5
        - idna:              3.4
        - imageio:           2.31.2
        - importlib-metadata: 6.7.0
        - importlib-resources: 5.12.0
        - ipykernel:         6.16.2
        - ipython:           7.34.0
        - ipython-genutils:  0.2.0
        - ipywidgets:        8.1.1
        - jedi:              0.19.0
        - jinja2:            3.1.2
        - joblib:            1.3.2
        - jsonschema:        4.17.3
        - jupyter:           1.0.0
        - jupyter-client:    7.4.9
        - jupyter-console:   6.6.3
        - jupyter-core:      4.12.0
        - jupyter-server:    1.24.0
        - jupyterlab-pygments: 0.2.2
        - jupyterlab-widgets: 3.0.9
        - kiwisolver:        1.4.5
        - llvmlite:          0.36.0
        - markdown:          3.4.4
        - markupsafe:        2.1.3
        - matplotlib:        3.5.3
        - matplotlib-inline: 0.1.6
        - mayavi:            4.8.1
        - mistune:           3.0.1
        - monoscene:         0.0.0
        - multidict:         6.0.4
        - nbclassic:         1.0.0
        - nbclient:          0.7.4
        - nbconvert:         7.6.0
        - nbformat:          5.8.0
        - nest-asyncio:      1.5.8
        - networkx:          2.6.3
        - notebook:          6.5.6
        - notebook-shim:     0.2.3
        - numba:             0.53.0
        - numpy:             1.20.3
        - nvidia-cublas-cu11: 11.10.3.66
        - nvidia-cuda-nvrtc-cu11: 11.7.99
        - nvidia-cuda-runtime-cu11: 11.7.99
        - nvidia-cudnn-cu11: 8.5.0.96
        - oauthlib:          3.2.2
        - omegaconf:         2.0.6
        - opencv-python:     4.5.1.48
        - packaging:         23.1
        - pandas:            1.3.5
        - pandocfilters:     1.5.0
        - parso:             0.8.3
        - pexpect:           4.8.0
        - pickleshare:       0.7.5
        - pillow:            9.5.0
        - pip:               22.3.1
        - pkgutil-resolve-name: 1.3.10
        - prometheus-client: 0.17.1
        - prompt-toolkit:    3.0.39
        - protobuf:          3.19.6
        - psutil:            5.9.5
        - ptyprocess:        0.7.0
        - pyasn1:            0.5.0
        - pyasn1-modules:    0.3.0
        - pycparser:         2.21
        - pydeprecate:       0.3.1
        - pyface:            8.0.0
        - pygments:          2.16.1
        - pyparsing:         3.1.1
        - pyqt5:             5.15.10
        - pyqt5-qt5:         5.15.2
        - pyqt5-sip:         12.13.0
        - pyrsistent:        0.19.3
        - python-dateutil:   2.8.2
        - pytorch-lightning: 1.4.9
        - pytz:              2023.3.post1
        - pywavelets:        1.3.0
        - pyyaml:            5.3.1
        - pyzmq:             24.0.1
        - qtconsole:         5.4.4
        - qtpy:              2.4.0
        - requests:          2.31.0
        - requests-oauthlib: 1.3.1
        - rsa:               4.9
        - scikit-image:      0.18.1
        - scikit-learn:      0.24.0
        - scipy:             1.7.3
        - send2trash:        1.8.2
        - setuptools:        65.6.3
        - six:               1.16.0
        - sniffio:           1.3.0
        - soupsieve:         2.4.1
        - tensorboard:       2.11.2
        - tensorboard-data-server: 0.6.1
        - tensorboard-plugin-wit: 1.8.1
        - terminado:         0.17.1
        - threadpoolctl:     3.1.0
        - tifffile:          2021.11.2
        - tinycss2:          1.2.1
        - torch:             1.13.1
        - torchmetrics:      0.6.0
        - torchvision:       0.14.1
        - tornado:           6.2
        - tqdm:              4.49.0
        - traitlets:         5.9.0
        - traits:            6.4.3
        - traitsui:          8.0.0
        - typing-extensions: 4.7.1
        - urllib3:           1.26.16
        - vtk:               9.2.6
        - wcwidth:           0.2.6
        - webencodings:      0.5.1
        - websocket-client:  1.6.1
        - werkzeug:          2.2.3
        - wheel:             0.38.4
        - widgetsnbextension: 4.0.9
        - yarl:              1.9.2
        - zipp:              3.15.0
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.7.16
        - release:           5.15.0-100-generic
        - version:           #110~20.04.1-Ubuntu SMP Tue Feb 13 14:25:03 UTC 2024

</details>

### More info

_No response_

**Comment by SkafteNicki** [comment]
Hi thanks for raising this issue. The issue you are seeing is something that can commonly happen in debuggers in general also vscode debugger. There are multiple reasons for why a tensor in debug view may only showcase as a tensor object:

* Lazy evaluation / hooks: VS Code sometimes shows Tensor(...) placeholders or stale values until execution resumes.
* CUDA tensors: Debuggers don’t always show GPU memory correctly; you might see ... or seemingly empty arrays even though the tensor is fine on device.
* In-place ops: If you step line by line and inspect a tensor that’s been modified in place (+=), the debugger may show it as “uninitialized” or all zeros, while printing it with print(tensor) in the terminal reveals the correct values.
* Autograd views: Debugger object inspectors don’t always resolve tensor.data or tensor.grad_fn properly.

Regardless of reason for the tensor not showing correctly in the debugger, this is not a issue in lightning and can be reproduced by running native torch. I am therefore going to close the issue for now.
